src/unstated.js

```diff
@@ -81,12 +83,26 @@ export class Subscribe<Containers: ContainersType> extends React.Component<
     }
 
     let safeMap = map;
-    let instances = containers.map(Container => {
-      let instance = safeMap.get(Container);
+    let instances = containers.map(ContainerItem => {
+      let instance: ContainerType;
+
+      if (
+        typeof ContainerItem === 'object' &&
+        ContainerItem instanceof Container
+      ) {
+        instance = ContainerItem;
+
+        let Class = instance.constructor;
+        if (!safeMap.has(Class)) {
+          safeMap.set(Class, instance);
```

Comment:
I kinda wonder if we should not add it to the global map. If you're manually injecting instances, you probably have a global instance you're always pulling from. I wonder if you can do anything interesting by inserting instances manually

Reply:
What if `<Subscribe>` had an `inject` prop too?

```jsx
<Subscribe to={[LoginContainer]} inject={[sharedAppContainer]}>
	…
</Subscribe>
```

Reply:
That would be difficult to type. I'm not sure how writing `(...to, ...inject) => Node` would be done.

I'm thinking it would be better to not add the instance to the global map

Reply:
I agree